samza-core/src/test/java/org/apache/samza/processor/TestStreamProcessor.java

```diff
@@ -43,53 +48,85 @@
 import static org.mockito.Mockito.when;
 
 public class TestStreamProcessor {
+  private Map<String, Boolean> processorListenerState;
+  private static final String ON_START = "onStart";
+  private static final String ON_SHUTDOWN = "onShutdown";
+  private static final String ON_FAILURE = "onFailure";
+
+  @Before
+  public void before() {
+    processorListenerState = new HashMap<String, Boolean>() {
+      {
+        put(ON_START, false);
+        put(ON_FAILURE, false);
+        put(ON_SHUTDOWN, false);
+      }
```

Comment:
I will stick with using the Map for now, as it seems *slightly* easier to read the code and assertions. If it gets too verbose, I will modify them to a set.